packages/token-bridge-sdk/src/hooks/arbTokenBridge.types.ts

```diff
@@ -48,24 +48,75 @@ export enum AssetType {
 
 export type TransactionLifecycle<Tx, TxReceipt> = Partial<{
   onTxSubmit: (tx: Tx) => void
-  onTxConfirm: (txReceipt: TxReceipt) => void
+  onTxConfirm: (tx: Tx, txReceipt: TxReceipt) => void
 }>
 
-export type L1EthDepositTransactionLifecycle = TransactionLifecycle<
-  L1EthDepositTransaction,
-  L1EthDepositTransactionReceipt
->
+export type TokenTransactionLifecycle<Tx, TxReceipt> = Pick<
+  TransactionLifecycle<Tx, TxReceipt>,
+  'onTxConfirm'
+> &
+  Partial<{
+    onTxSubmit: (tx: Tx, symbol: string) => void
+  }>
 
-export type L1ContractCallTransactionLifecycle = TransactionLifecycle<
-  L1ContractCallTransaction,
-  L1ContractCallTransactionReceipt
->
+export type L1EthDepositTransactionLifecycle = Pick<
+  TransactionLifecycle<L1EthDepositTransaction, L1EthDepositTransactionReceipt>,
+  'onTxSubmit'
+> & {
+  onTxConfirm: (
+    tx: L1EthDepositTransaction,
+    txReceipt: L1EthDepositTransactionReceipt,
+    ethDepositMessage: EthDepositMessage
+  ) => void
+}
```

Comment:
A couple of things here (will probably relate to the other lifecycle types, too):

* I think I may have overcomplicated things with the types, so maybe we should just repeat ourselves a bit, should make things more readable (later we can figure out a better abstraction)
* Let's call the lifecycle methods with objects, so we have named parameters
* Let's rename the `onConfirm` to `onL1TxSuccess`, and add `onL1TxFailure` and `onEthDepositMessage` hooks

```suggestion
export type EthDepositTransactionLifecycle = {
  onL1TxSubmit: ({ 
    tx 
  }: { 
    tx: L1EthDepositTransaction 
  }) => void
  onL1TxSuccess: ({ 
    tx, 
    txReceipt 
  }: { 
    tx: L1EthDepositTransaction, 
    txReceipt: L1EthDepositTransactionReceipt 
  }) => void
  onL1TxFailure: ({ 
    tx, 
    txReceipt 
  }: { 
    tx: L1EthDepositTransaction, 
    txReceipt: L1EthDepositTransactionReceipt 
  }) => void
  onEthDepositMessage: ({
    ethDepositMessage
  }: {
    ethDepositMessage: EthDepositMessage
  }) => void
}
```